Modules/Preview/VixenPreview/OpenGL/OpenGLPreviewForm.cs

```diff
@@ -634,9 +641,9 @@ private void DrawPoints(Matrix4 mvp)
 				_program["mvp"].SetValue(mvp);
 				_program["pointScale"].SetValue(_pointScaleFactor);
 
-				foreach (var dataDisplayItem in Data.DisplayItems)
+				foreach (var dataDisplayItem in Data.DisplayItems.Where(item => item.IsLightShape()))
 				{

```

Comment:
I wonder about these places that are having to determine if it is a light shape in these update drawing routines if it would be best to sort them out into two lists when the form is initialized instead of doing that where logic over and over. Those lists only change when the preview is edited and it will be reloaded when that happens. These areas are called very often and Linq queries can have allocation and performance costs that could be alleviated.

Reply:
I attempted to update the code per your suggestion.  I saw two spots in this file where it looked like we would see performance improvements using a cached collection.  Let me know if you see others.

Reply:
Addressing  your comment about serialization, my profile reads in fine.  I think the XML has the concrete class listed in it.  I generally feel that this ticket contains some risk because it is not really feasible for me to manually test all the code paths I touched.

Reply:
The feature branch will give us opportunities to do extra validation on that before we expose it to the mainstream.

Reply:
I pushed the change to wrong branch.  I have now applied the change to this branch.